index.js

```diff
@@ -31,64 +31,62 @@ if(!heimdall.hasMonitor('babel')) {
   });
 }
 
-module.exports = Babel;
-function Babel(inputTree, _options) {
-  if (!(this instanceof Babel)) {
-    return new Babel(inputTree, _options);
-  }
-
-  let options = _options || {};
-  options.persist = 'persist' in options ? options.persist : true;
-  options.async = true;
-
-  if (options.browserPolyfill) {
-    let polyfillPath = require.resolve('@babel/polyfill/package.json');
-    polyfillPath = path.join(path.dirname(polyfillPath), "dist");
-
-    let browserPolyfillPath = options.browserPolyfillPath || "browser-polyfill.js"
-    let polyfill = funnel(polyfillPath, {
-      files: ['polyfill.js'],
-      getDestinationPath: () => browserPolyfillPath
-    });
-    inputTree = mergeTrees([polyfill, inputTree]);
-  }
-  delete options.browserPolyfill;
-  delete options.browserPolyfillPath;
+class Babel extends Filter {
+  constructor(inputTree, _options = {}) {
+    let options = _options;
+    options.persist = 'persist' in options ? options.persist : true;
+    options.async = true;
+
+    if (options.browserPolyfill) {
+      let polyfillPath = require.resolve('@babel/polyfill/package.json');
+      polyfillPath = path.join(path.dirname(polyfillPath), "dist");
+
+      let browserPolyfillPath = options.browserPolyfillPath || "browser-polyfill.js"
+      let polyfill = funnel(polyfillPath, {
+        files: ['polyfill.js'],
+        getDestinationPath: () => browserPolyfillPath
+      });
+      inputTree = mergeTrees([polyfill, inputTree]);
+    }
```

Comment:
Since this is going to be a major version bump, I'd generally prefer to drop support for the `browserPolyfill` option. Seems totally fine to do in a follow up, but I don't want to release 8.x without dropping it...

Created https://github.com/babel/broccoli-babel-transpiler/issues/188 to track that.